### PR TITLE
New package libacvp

### PR DESCRIPTION
**Pull request by xnox**
A useful tool for testing OpenSSL against ACVP Demo server.

### CODE REVIEW

**Review by smoser — CHANGES_REQUESTED** [review]
Can you add test to execute `acvp_app`?

It looks like it has useful information in `--version`

```
# acvp_app  --version

 ACVP library version: libacvp_oss-2.1.2
ACVP protocol version: 1.0

 Compiled SSL version: OpenSSL 3.3.2 3 Sep 2024
   Linked SSL version: OpenSSL 3.3.2 3 Sep 2024

       FIPS requested: yes
FIPS Provider Version: (null)

# echo $?
1
```

not so helpfully it exits 1.

**Comment by xnox** [comment]
> Can you add test to execute `acvp_app`?
> 
> It looks like it has useful information in `--version`
> 
> ```
> # acvp_app  --version
> 
>  ACVP library version: libacvp_oss-2.1.2
> ACVP protocol version: 1.0
> 
>  Compiled SSL version: OpenSSL 3.3.2 3 Sep 2024
>    Linked SSL version: OpenSSL 3.3.2 3 Sep 2024
> 
>        FIPS requested: yes
> FIPS Provider Version: (null)
> 
> # echo $?
> 1
> ```
> 
> not so helpfully it exits 1.

i guess i can propose this package to enterprise-packages and then test will pass, as i will be able to depend on a fips module in the test.

**Comment by xnox** [comment]
even with fips provider it exists 1, lol =) so will structure a tight test for this.

File: libacvp.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+subpackages:
+  - name: ${{package.name}}-dev
+    pipeline:
+      - uses: split/dev
+
+test:
+  pipeline:
+    - runs: |
+        # Always exits 1, see https://github.com/cisco/libacvp/issues/872
+        ! acvp_app --version >version.output
```
Comment (smoser) [inline libacvp.yaml:45]: I like it. thank you.

**Review by smoser — APPROVED** [review]
this is good.  some update version will fix --version to exit 0.
that update PR will fail and someone will change this to accomodate.
thanks.

**Pull request merged**